cmd/datagen/datagen.go

```diff
@@ -0,0 +1,201 @@
+*/
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/klauspost/compress/zstd"
+	"github.com/sams96/rgeo"
+	"github.com/twpayne/go-geom/encoding/geojson"
+)
+
+func main() {
+	outPath := flag.String("o", "", "path to output file")
+	propsFilePath := flag.String("merge", "", "path to file to merge properties from")
+	flag.Parse()
+
+	if *outPath == "" {
+		_, _ = fmt.Fprintf(os.Stderr, "usage: %s -o outprefix <infile.geojson> [infile2.geojson] [...]\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	inputFiles := flag.Args()
+
+	// cosmetics for generating the .txt attribution file
+	attributionFiles := append([]string{}, inputFiles...)
+	if *propsFilePath != "" {
+		attributionFiles = append(attributionFiles, *propsFilePath)
+	}
+	for i, path := range attributionFiles {
+		attributionFiles[i] = filepath.Base(path)
+	}
+
+	if fc, err := readInputs(inputFiles, *propsFilePath); err != nil {
+		log.Fatal("error reading inputs: ", err)
+	} else if err := writeFeatures(*outPath, *fc); err != nil {
+		log.Fatal("error writing features: ", err)
+	} else if err := writeAttribution(*outPath, attributionFiles); err != nil {
+		log.Fatal("error writing attribution: ", err)
+	}
+}
+
+func readInputs(in []string, propsFileName string) (*geojson.FeatureCollection, error) {
+	var props *geojson.FeatureCollection
+	if propsFileName != "" {
+		md, err := readGeoJSON(propsFileName)
+		if err != nil {
+			return nil, fmt.Errorf("read props GeoJSON file: %w", err)
+		}
+		props = md
+	}
+
+	fc := &geojson.FeatureCollection{}
+	for _, f := range in {
+		s, err := readGeoJSON(f)
+		if err != nil {
+			return nil, fmt.Errorf("read input GeoJSON file: %w", err)
+		}
+		if props != nil {
+			if err := extendProps(s, props); err != nil {
+				return nil, fmt.Errorf("extend properties: %w", err)
+			}
+		}
+		fc.Features = append(fc.Features, s.Features...)
+	}
+
+	return fc, nil
+}
+
+func writeFeatures(outPath string, fc geojson.FeatureCollection) error {
+	f, err := os.Create(outPath)
+	if err != nil {
+		return fmt.Errorf("create output file: %w", err)
+	}
+	defer func() { _ = f.Close() }()
+
+	zw, err := zstd.NewWriter(f, zstd.WithEncoderLevel(zstd.SpeedBestCompression))
+	if err != nil {
+		return fmt.Errorf("create zstd writer: %w", err)
+	}
+	defer func() { _ = zw.Close() }()
+
+	dataset, err := rgeo.LoadGeoJSON(fc)
+	if err != nil {
+		return fmt.Errorf("load GeoJSON: %w", err)
+	}
+
+	if err := dataset.Encode(zw); err != nil {
+		return fmt.Errorf("encode dataset: %w", err)
+	}
+
+	// explicit flush so that zw.Close always succeeds
+	if err := zw.Flush(); err != nil {
+		return fmt.Errorf("flush: %w", err)
+	}
+
+	return nil
+}
+
+// readGeoJSON parses a GeoJSON file as geojson.FeatureCollection
+func readGeoJSON(path string) (*geojson.FeatureCollection, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("open file: %w", err)
+	}
+	defer func() {
+		_ = f.Close()
+	}()
```

Comment:
lets condense that into a single line, like in every other `defer`'ed `Close()`

Reply:
I just meant a cosmetic change from the above 3 lines to the single line

`defer func() { _ = f.Close() }()`